compiler/resolution/lowerIterators.cpp

```diff
@@ -1562,28 +1568,39 @@ static void processShadowVariables(ForLoop* forLoop, SymbolMap *map) {
 
         VarSymbol* capturedSvar = new VarSymbol(astr("cap_", svar->name), svar->type);
         forLoop->insertBefore(new DefExpr(capturedSvar));
-        forLoop->insertBefore(new CallExpr(PRIM_MOVE, capturedSvar, svar->outerVarSE->symbol()));
+
+        CallExpr *initMove = toCallExpr(svar->initBlock()->body.first());
+        SymbolMap map1;
+        map1.put(svar->ParentvarForIN(), svar->outerVarSE->symbol());
+        Expr *copiedInitialization = initMove->get(2)->copy(&map1);
+        forLoop->insertBefore(new CallExpr(PRIM_MOVE, capturedSvar, copiedInitialization));
+
+        SymbolMap map2;
+        map2.put(svar->ParentvarForIN(), capturedSvar);
+        Expr *copiedInitialization2 = initMove->get(2)->copy(&map2);
+        (void)copiedInitialization2;
```

Comment:
cruft?

Reply:
Yeah...

I'll admit this whole section of code is a bit of a kludge for me to make a copy of the shadow'd variable.

I'll add in this comment that explains somewhat, how I'm going on about that:

```
  // Shadow variables have an "initBlock", we want to use this block to
  // get a copy of the variable. An example of svar->initBlock() might
  // look like this:
  //
  //  (BlockStmt
  //    (CallExpr move
  //      (SymExpr 'const-val this')
  //        (CallExpr
  //          (SymExpr 'fn chpl__initCopy')
  //          (SymExpr 'const-val INP_this')))
  //
  // But we rather than getting a copy of INP_this we want to get a copy of
  // the outer variable that the shadow variable is shadowing (i.e.
  // outerVarSE).
```

Which I think might raise the question what is this `INP_` variable. And I'm actually not sure there, it looks like for some given the shadow variables `var` we model `INP_var = var` and then have another shadow variable `shadow_var = INP_var` (where really `shadow_var` is just directly named `var`). I think this modeling may be accomplishing for `forall` looks kind of what I'm doing with this whole dance about getting a "captured" copy of the variable. But I'm really not sure.

Anyway, all this is to say is that there might be an opportunity to something more intelligent (or atleast consistent with what we do for forall loops) than what I'm doing here but what I have does seem to work.